src/core/net/nat64_translator.cpp

```diff
@@ -596,16 +617,28 @@ void Translator::ProtocolCounters::Count4To6Packet(uint8_t aProtocol, uint64_t a
     mTotal.m4To6Bytes += aPacketSize;
 }
 
-State Translator::GetState(void) const
+void Translator::UpdateState()
 {
-    State ret = kStateDisabled;
+    State newState;
 
-    VerifyOrExit(mEnabled);
-    VerifyOrExit(mIp4Cidr.mLength > 0 && mNat64Prefix.IsValidNat64(), ret = kStateNotRunning);
-    ret = kStateActive;
+    if (mEnabled)
+    {
+        if (mIp4Cidr.mLength > 0 && mNat64Prefix.IsValidNat64())
+        {
+            newState = kStateActive;
+        }
+        else
+        {
+            newState = kStateNotRunning;
+        }
+    }
+    else
+    {
+        newState = kStateDisabled;
+    }
 
-exit:
-    return ret;
+    Get<Notifier>().Update(mState, newState, kEventNat64TranslatorStateChanged);
```

Comment:
May be better doing `SuccessOrExit(Get<No...)` to avoid logging when there is no change in state.

Reply:
Done